### PR TITLE
Add endpoint for cloud subnets

### PR DESCRIPTION
**Pull request by lfu**
https://github.com/ManageIQ/manageiq-api/issues/1013

@miq-bot add_label enhancement, kasparov/no

### CODE REVIEW

**Comment by lfu** [comment]
```
POST /api/cloud_subnets
{
         "action" : "create",
         "resource" : {
           "ems_id" : 10000000000013,
           "name"   : "test_cloud_subnet"
         }
}

{
    "results": [
        {
            "success": true,
            "message": "Creating Cloud Subnet test_cloud_subnet for Provider: OpenStack Network Manager",
            "task_id": "10000000053334",
            "task_href": "http://localhost:3000/api/tasks/10000000053334"
        }
    ]
}
```

```
POST /api/cloud_subnets/10000000000028
{
         "action": "edit",
         "status": "available"
}

{
    "success": true,
    "message": "Updating Cloud Subnet id: 10000000000028 name: 'provider'",
    "task_id": "10000000053335",
    "task_href": "http://localhost:3000/api/tasks/10000000053335"
}
```

**Review by agrare — APPROVED** [review]

**Comment by agrare** [comment]
ping @abellotti @bdunne

File: app/controllers/api/cloud_subnets_controller.rb [hunk 1]
```diff
@@ -15,5 +15,42 @@ def options
 
       render_options(:cloud_subnets, :form_schema => klass.params_for_create(ems))
     end
+
+    def create_resource(_type, _id = nil, data = {})
+      ems = ExtManagementSystem.find(data['ems_id'])
+      klass = CloudSubnet.class_by_ems(ems)
+      raise BadRequestError, "Create cloud subnet for Provider #{ems.name}: #{klass.unsupported_reason(:create)}" unless klass.supports?(:create)
```
Comment (abellotti) [inline app/controllers/api/cloud_subnets_controller.rb:22]: shouldn't this be Cannot create cloud subnet ...  same for the Update and Delete BadRequestError

**Review by abellotti — APPROVED** [review]

**Comment by NickLaMuro** [comment]
Closing in favor of https://github.com/ManageIQ/manageiq-api/pull/1025 (which fixes the tests that are failing here)

**Pull request closed without merging**